File: src/leetcode/python/ConstructTheRectangle_492.py
import math


class ConstructTheRectangle_492(object):
    def constructRectangle(self, area):
        """
        :type area: int
        :rtype: List[int]
        """
        limit = int(math.sqrt(area))
        while area % limit != 0:
            limit -= 1
        return [area / limit, limit]

File: src/leetcode/java/ConstructTheRectangle_492.java
package leetcode.java;

public class ConstructTheRectangle_492 {
  public int[] constructRectangle(int area) {
    int fir = 0, sec = 0;
    int limit = (int) Math.sqrt(area);
    while (area % limit != 0) {
      limit--;
    }
    fir = area / limit;
    sec = limit;
    return new int[] { fir, sec };
  }
}
